contentcuration/contentcuration/frontend/shared/app.js

```diff
@@ -43,20 +45,34 @@ Vue.component('ActionLink', ActionLink);
 
 export let rootVue;
 
-export default function startApp({ store, router, index }) {
-  initializeDB().then(() => {
-    const config = {
-      el: 'app',
-      store,
-      router,
-    };
-
-    if (index) {
-      Object.assign(config, index);
-    } else {
-      Object.assign(config, Base);
-    }
-
-    rootVue = new Vue(config);
-  });
+export default async function startApp({ store, router, index }) {
+  await initializeDB();
+
+  const currentUser = window.user || {};
+  const dbCurrentUser = (await Session.get(CURRENT_USER)) || {};
+
+  if (
+    currentUser.id === undefined ||
```

Comment:
I think we can simplify this conditional? The main thing is making sure if the previously logged in user was not a guest, and now the current user is a different user?

```
dbCurrentUser.id && dbCurrentUser.id !== currentUser.id
```